### PR TITLE
added all filters

### PR DESCRIPTION
**Pull request by varnita619**
added 
- filter by price range
- filter by category
- filter by rating
- sort by price high to low or
- price low to high
- clear all filters

Screenshot (19)

### CODE REVIEW

File: src/Components/SideBar/SideBar.jsx [hunk 1]
```diff
@@ -1,87 +1,161 @@
...
+              <input
+                type="radio"
+                id="4Stars"
+                name="radio"
+                value="4"
+                checked={rating === "4"}
+                onChange={(e) =>
+                  dispatch({ type: "SORT_BY_RATING", payload: e.target.value })
+                }
+              />{" "}
```
Comment (anuj-kumary) [inline src/Components/SideBar/SideBar.jsx:73]: Remove this braces

File: src/Context/ProductsContext.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import { createContext, useContext, useEffect, useReducer } from "react";
+import { productsReducerFunction } from "../Reducer/ProductsReducer.js";
+import axios from "axios";
+import { filterByPriceRange } from "../Utils/filterByPriceRange.js";
+import { sortByPrice } from "../Utils/sortByPrice.js";
+import { filterByCategory } from "../Utils/filterByCategory.js";
+import { filterByRating } from "../Utils/filterByRating.js";
+import { uniqueCategory } from "../Utils/uniqueCategory.js";
```
Comment (anuj-kumary) [inline src/Context/ProductsContext.jsx:8]: You can use here barrel import

**Review by anuj-kumary — APPROVED** [review]
Good use of naming conventions also liked your folder structure.

File: src/CSS/Reset.css [hunk 3]
```diff
@@ -2,6 +2,7 @@
     --primary-color:black;
     --secondary-color:rgba(21, 21, 21, 0.5);
     --white:white;
+    --background-color:rgba(209, 209, 209, 0.993);
     --btn-hover:rgba(238, 231, 231, 0.527);
     --subheading:rgba(21, 21, 21, 0.5);
     --danger-color:red;
```
Comment (pritam-kr) [inline src/CSS/Reset.css]: use hex colors

File: src/Reducer/ProductsReducer.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                ...state,
+                sortBy: action.payload
+            };
+        case 'LOW_TO_HIGH':
+            return {
+                ...state,
+                sortBy: action.payload
+            };
+        case 'GET_CATEGORY':
+            if (state.category.includes(action.payload)) {
```
Comment (pritam-kr) [inline src/Reducer/ProductsReducer.js:24]: If possible make a file for only ACTIONS. and use it as a variable

File: src/Utils/filterByOutOfStock.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+function filterByOutOfStock(data, outOfStock){
+    console.log([...data].filter((item) => (outOfStock ? true : item.inStock)))
+    if(outOfStock === true){
+        return [...data].filter((item) => (outOfStock ? true : item.inStock));
+    }
+    
+    // console.log(outOfStock);
```
Comment (pritam-kr) [inline src/Utils/filterByOutOfStock.js:7]: remove unusual comment before doing PR

**Review by pritam-kr — APPROVED** [review]
Hey, I have added some points. you can look into it. and use code formatter.

**Pull request merged**